src/api/hooks/useGetANS.ts

```diff
@@ -26,43 +27,53 @@ function getFetchNameUrl(
 
 export function useGetNameFromAddress(address: string, shouldCache = false) {
   const [state, _] = useGlobalState();
-  const [name, setName] = useState<string | undefined>();
 
-  useEffect(() => {
-    const cachedName = getLocalStorageWithExpiry(address);
-    if (cachedName) {
-      setName(cachedName);
-      return;
-    }
+  const cachedName = getLocalStorageWithExpiry(address);
+  if (cachedName) {
+    return cachedName;
+  }
+
+  const queryResult = useQuery<string, ResponseError>({
+    queryKey: ["ANSName", address, shouldCache, state.network_name],
+    queryFn: () => genANSName(address, shouldCache, state.network_name),
+  });
 
-    const primaryNameUrl = getFetchNameUrl(state.network_name, address, true);
-    if (primaryNameUrl !== undefined) {
-      const fetchData = async () => {
-        const {name: primaryName} = await fetchJsonResponse(primaryNameUrl);
-
-        if (primaryName) {
-          if (shouldCache) {
-            setLocalStorageWithExpiry(address, primaryName, TTL);
-          }
-          setName(primaryName);
-        } else {
-          const nameUrl =
-            getFetchNameUrl(state.network_name, address, false) ?? "";
-          const {name} = await fetchJsonResponse(nameUrl);
-          if (shouldCache) {
-            setLocalStorageWithExpiry(address, name, TTL);
-          }
-          setName(name);
-        }
-      };
-
-      fetchData().catch((error) => {
-        console.error("ERROR!", error, typeof error);
-      });
+  // return empty string if ans name not found to prevent useQuery complaining about undefined return
+  return queryResult.data === "" ? undefined : queryResult.data;
```

Comment:
I don't quite get this. Why are we doing this? This also doesn't seem to do what the comment says, aka returning an empty string?

Reply:
undefined's not an accepted value to be returned from the query function. pls checkout this link below:

https://tanstack.com/query/v4/docs/react/guides/migrating-to-react-query-4#undefined-is-an-illegal-cache-value-for-successful-queries

so instead i have to make it return empty strings if ans name not found.

Reply:
how about `null` then?

Reply:
regardless the comment above is confusing above the return statement. If you put this comment above the `genANSName` it makes more sense. Also would be good to add the link to the tanstack query docs in the code for full context.
Regardless I think it's better to return null instead of empty string.